backend/blog/views.py

```diff
@@ -21,7 +20,9 @@ def blog_post(request, slug):
     """
     Single blog post view
     """
-    post = BlogPost.objects.get(slug=slug)
-    tags = Tag.objects.all()
-    context = {'post': post, 'tags': tags}
+    posts = BlogPost.objects.filter(slug=slug)
+    if not posts or (not posts[0].published and request.user != posts[0].author):
```

Comment:
This is a great direction, but let's make this even simpler. If user is signed in with blog privileges, they can see the post. The reason I think this rather than just author: students might want to share their unpublished posts with Catherine or someone else. Certainly sharing a URL will be easier than telling them where to go to the admin panel to see their work in progress. How does that sound?

Reply:
something like that! If they don't have permissions then send them to 404, otherwise send them the blog page (with the banner)